src/test/java/com/znsio/teswiz/runner/AppPathTest.java

```diff
@@ -0,0 +1,105 @@
+package com.znsio.teswiz.runner;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AppPathTest {
+    private static final String tempFolderPath = System.getProperty("user.dir") + File.separator + "temp";
+    private static final String expectedDirectoryPath = tempFolderPath + File.separator + "sampleApps";
+    private static final String fileName = "VodQA.apk";
+    private static final String expectedAppPath = expectedDirectoryPath + File.separator + fileName;
+    private static final String appPathAsUrl = "https://github.com/anandbagmar/sampleAppsForNativeMobileAutomation/raw/main/VodQA.apk";
+    private static final String appPathAsIncorrectUrl = "https://github.com/anandbagmar/sampleAppsForNativeMobileAutomation/ra/main/VodQA.apk";
+    private static final String appPathAsFilePath = expectedAppPath;
+    private static final String appPathAsIncorrectFilePath = tempFolderPath + File.separator + "smleApps" + File.separator + fileName;
+
+    @Test
+    void validateAppPathAsCorrectUrlCreateRepoAndDownloadFileWhenBothDoNotExist() {
+        deleteDirectorySubDirectoryAndFiles(tempFolderPath);
+        String actualAppPath = DeviceSetup.convertAppPathToFilePathIfNeeded(appPathAsUrl);
+        assertEquals(expectedAppPath, actualAppPath);
+        assertTrue(Files.exists(Paths.get(actualAppPath)));
+    }
+
+    @Test
+    void validateAppPathAsCorrectUrlDoNotDownloadFileWhenRepoAndFileAlreadyExist() {
+        String actualAppPath = DeviceSetup.convertAppPathToFilePathIfNeeded(appPathAsUrl);
+        assertEquals(expectedAppPath, actualAppPath);
+        assertTrue(Files.exists(Paths.get(actualAppPath)));
+    }
+
+    @Test
+    void validateAppPathAsCorrectUrlOnlyDownloadFileWhenRepoExist() {
+        deleteFile(expectedAppPath);
+        createDirectory(expectedDirectoryPath);
+        String actualAppPath = DeviceSetup.convertAppPathToFilePathIfNeeded(appPathAsUrl);
+        assertEquals(expectedAppPath, actualAppPath);
+        assertTrue(Files.exists(Paths.get(actualAppPath)));
+    }
+
+    @Test
+    void validateAppPathAsIncorrectUrlGivesError() {
```

Comment:
It would be better if we could specify the error thrown in the test name eg validateRuntimeErrorWithIncorrectAppPath

Reply:
done